students/DimaSakovich/04-nlp-full-circle/project_metrics.ipynb

```diff
@@ -0,0 +1,455 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "There are 2(3) parts of my course project:\n",
+    "\n",
+    "* grapheme-to-phoneme\n",
+    "* speech-to-text\n",
+    "* (questin answering as a bonus)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### speech2text metric"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The most popular metric for evaluating s2t models is WER"
```

Comment:
I'd like to see a comparison with other popular metrics (ROUGE, BLEU, METEOR) :)
Also, what about using Wacc? It's more natural for NLP to try to get to 1 instead of 0 :))

Reply:
было много работы на работе + кое какие обстоятельств, мешающие сфокусироваться на задаче, а выходные выпали из-за хакатона. Да и как-то, несколько раз к этой задаче подходил, но все никак не мог настроится и придумать как сделать ее правильно. Вот бывает такое, вроде бы простая задача, и понимаешь что ничего сложного нет, но вот не идет она тебе и все :(
Дай мне какой-то план по этой задаче, я реализую по твоим подсказкам, если так можно

Reply:
> I'd like to see a comparison with other popular metrics (ROUGE, BLEU, METEOR) :)
> Also, what about using Wacc? It's more natural for NLP to try to get to 1 instead of 0 :))
this metrics more appropriate for machine translation and I've seen any of them in articles related to sp2t domain. But, I can add them, just for comparison. May we firtsly discuss it after today's lecture?

Reply:
По фактам: я би нічого зараз особливо не доробляв, оскільки є багато інших завдань. Основна ідея була пройти повний цикл NLP-проєкту. Я, просто, перепитував, щоб зрозуміти: що, власне, в самій задачі викликало складність. Тепер ясно, що проблема була не в ній :) Єдине, що точно треба доробити — це показати, які дані було отримано в результаті застосування правил.
По курсовому: я би хотів побачити хоча б базовий порівняльний аналіз з BLEU/... - чому саме WER беруть у t2s. До речі, на днях бачив моджливо цікаву статтю на тему розробок нових метрик: https://arxiv.org/abs/2004.04696 Ще одне питання: чому WER гарно підходить для графем? Може, edit distance був би кращий?

Reply:
Ок, тогда я доделываю по домашке вывод результата правил, а по проектным метрикам делаю сравнительный анализ всех схожих метрик